server/task-distributed/src/main/java/org/apache/james/task/eventsourcing/distributed/RabbitMQExclusiveConsumer.java

```diff
@@ -0,0 +1,170 @@
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Function;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.ImmutableMap;
+import com.rabbitmq.client.CancelCallback;
+import com.rabbitmq.client.Channel;
+import com.rabbitmq.client.Connection;
+import com.rabbitmq.client.DeliverCallback;
+import com.rabbitmq.client.impl.recovery.AutorecoveringConnection;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.FluxSink;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Scheduler;
+import reactor.core.scheduler.Schedulers;
+import reactor.rabbitmq.AcknowledgableDelivery;
+import reactor.rabbitmq.ConsumeOptions;
+import reactor.rabbitmq.RabbitFluxException;
+import reactor.rabbitmq.ReceiverOptions;
+
+/**
+ * @deprecated to remove once the parallel execution of task has been implemented
+ */
+@Deprecated
+public class RabbitMQExclusiveConsumer implements Closeable {
+    private static final Function<Connection, Channel> CHANNEL_CREATION_FUNCTION = new RabbitMQExclusiveConsumer.ChannelCreationFunction();
+    private static final Logger LOGGER = LoggerFactory.getLogger(RabbitMQExclusiveConsumer.class);
+
+    private static final boolean NON_LOCAL = true;
+    private static final boolean EXCLUSIVE = true;
+
+    private static class ChannelCreationFunction implements Function<Connection, Channel> {
+
+        @Override
+        public Channel apply(Connection connection) {
+            try {
+                return connection.createChannel();
+            } catch (IOException e) {
+                throw new RabbitFluxException("Error while creating channel", e);
+            }
+        }
+    }
+
+    private Mono<? extends Connection> connectionMono;
+    private final AtomicBoolean hasConnection;
+    private final Scheduler connectionSubscriptionScheduler;
+    private final boolean privateConnectionSubscriptionScheduler;
+
+    public RabbitMQExclusiveConsumer(ReceiverOptions options) {
+        this.privateConnectionSubscriptionScheduler = options.getConnectionSubscriptionScheduler() == null;
+        this.connectionSubscriptionScheduler = options.getConnectionSubscriptionScheduler() == null ?
+            createScheduler("rabbitmq-receiver-connection-subscription") : options.getConnectionSubscriptionScheduler();
+        hasConnection = new AtomicBoolean(false);
+        this.connectionMono = options.getConnectionMono() != null ? options.getConnectionMono() :
+            Mono.fromCallable(() -> options.getConnectionFactory().newConnection())
+                .doOnSubscribe(c -> hasConnection.set(true))
+                .subscribeOn(this.connectionSubscriptionScheduler)
+                .cache();
+    }
+
+    protected Scheduler createScheduler(String name) {
+        return Schedulers.newElastic(name);
+    }
+
+
+    public Flux<AcknowledgableDelivery> consumeExclusiveManualAck(final String queue, ConsumeOptions options) {
```

Comment:
s/final String queue/String queue/

Reply:
What's that String? How about strong typing?

Reply:
Is that really a 60 line long + 7 indentation level method you ask me to review???

Reply:
TLDR. Can you explain with simple english the mechanisms and the algorithm you used to achieve exclusive consume?

Reply:
idem

Reply:
We didn't use complicated algorithms, we only set the boolean parameter `exclusive` to `true` on the channel.basicConsume method.
That's why we were forced to copy and adapt the reactor rabbit method. Because there were no possibility to provide this parameter in the `Receiver` code